### DISCUSSION TITLE
Link prediction on multiple graph with custom negative edge sampling

### DISCUSSION THREAD

**Discussion by GianlucaDeStefano** [pyg-team/pytorch_geometric#6151] [Q&A] [answered]
Hi all, I have recently been migrating to Pytorch Geometric from another library and am curious about how I should handle the following use case.

I have multiple separate graphs and want to train a link prediction model.
Each of these graphs is logically disconnected from the others. I want to avoid sampling negative edges that would form a bridge between any two of these graphs; as for my problem, it would be illogical and would actually damage the results.

In PyTorch geometric, what is the best way of managing these graphs for my task? I was looking at creating my custom dataset class and using diagonal batching. Is this the best way?

Furthermore, is there any prebuilt functionality to guarantee the sampling-of-negative-edges described above?

Thank you for any input you may give me,
Gianluca

**Comment by rusty1s** [comment, marked as answer]
I think you can use the normal `DataLoader` for this to batch your multiple graphs together, but just need to make sure to not sample negative links between two different graphs. There is a `utils.batched_negative_sampling` functionality which you can call inside your training loop to achieve this. Hope this helps!
